src/controller/CHIPDeviceController.cpp

```diff
@@ -3207,6 +3223,39 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
         }
     }
     break;
+    case CommissioningStage::kConfigureTCAcknowledgments: {
+        ChipLogProgress(Controller, "Setting Terms and Conditions");
+
+        if (!params.GetRequireTermsAndConditionsAcknowledgement())
+        {
+            ChipLogProgress(Controller, "Setting Terms and Conditions: Skipped");
+            CommissioningStageComplete(CHIP_NO_ERROR);
+            return;
+        }
+
+        if (!params.GetTermsAndConditionsAcknowledgement().HasValue())
+        {
+            ChipLogError(Controller, "No acknowledgements provided");
+            CommissioningStageComplete(CHIP_ERROR_INCORRECT_STATE);
+            return;
+        }
+
+        GeneralCommissioning::Commands::SetTCAcknowledgements::Type request;
+        TermsAndConditionsAcknowledgement termsAndConditionsAcknowledgement = params.GetTermsAndConditionsAcknowledgement().Value();
+        request.TCUserResponse = termsAndConditionsAcknowledgement.acceptedTermsAndConditions;
+        request.TCVersion      = termsAndConditionsAcknowledgement.acceptedTermsAndConditionsVersion;
+
+        ChipLogProgress(Controller, "Setting Terms and Conditions: %hu, %hu", request.TCUserResponse, request.TCVersion);
+        CHIP_ERROR err =
+            SendCommissioningCommand(proxy, request, OnSetTCAcknowledgementsResponse, OnBasicFailure, endpoint, timeout);
```

Comment:
Follow-up: consider having a separation that allows commissioning API callers to seek the correct data to use based on their external access to DCL, and continue commissioning when the terms provided by users have been received. Current impl requires this to be known a-priori, but the device being commissioned may not be known a-priori since many flows (e.g. manual pairing code) start without knowing the device type involved.

Reply:
I had this in the previous revision, but... Boris asked to remove it, due to unnecessary complexity. Also... there is more to the story. Here is the comment I provided above.

> I was originally under the impression that we supported 11-digit code that did not provide VID/PID then identified the VID/PID from an announcement from the device, and then could perform a DCL lookup. I learned that we only support 21-digit and QR code for custom commissioning flow (from Chris D.). So…. We should know the VID/PID prior to initiating the AutoCommissioner, and we can require the UX to provide the acknowledgement as a pre-condition, thus avoiding any “wait” stage.